sdk/spec_file_generator.go

```diff
@@ -0,0 +1,37 @@
+package sdk
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+)
+
+type protocol string
+
+const (
+	protoTCP       protocol = "tcp"
+	protoNamedPipe protocol = "npipe"
+)
+
+func writeSpec(name, address string, proto protocol) (string, error) {
+
+	var pluginSpecDir string
+	if runtime.GOOS == "windows" {
+		pluginSpecDir = ([]string{filepath.Join(os.Getenv("programdata"), "docker", "plugins")})[0]
```

Comment:
This concerns me. It should follow the data-root of the daemon. `programdata\docker` is created by Docker with a specific ACL to allow two daemons to run side-by-side (such as under docker CI) as it is not possible to run DinD on Windows. If there is any possibility that a plug-in can create this directory without the ACL before a daemon has started, the ACL would be default here.